tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -445,18 +469,31 @@ StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStream(
             << " @ index: " << index.ToString();
 
     if (output_info.alias_config) {
-      ExecutionInput& input = arguments[allocation->parameter_number()];
       MaybeOwningDeviceMemory* maybe_owning_memory =
-          input.MutableBuffer(allocation->param_shape_index());
+          [&]() -> xla::MaybeOwningDeviceMemory* {
+        // ScopedBuffer is never an owned buffer.
+        if (auto* unowned_shapedbuffers =
+                absl::get_if<absl::Span<const ShapedBuffer* const>>(
+                    &arguments)) {
+          return nullptr;
+        } else {
+          auto unowned_execution_input =
+              absl::get<absl::Span<ExecutionInput>>(arguments);
+          ExecutionInput& input =
+              unowned_execution_input[allocation->parameter_number()];
+          return input.MutableBuffer(allocation->param_shape_index());
+        }
+      }();
       if (output_info.alias_config->must_alias() &&
-          !maybe_owning_memory->HasOwnership()) {
+          !maybe_owning_memory) {
```

Comment:
It is this change that introduced the bug. The good code should be:
`maybe_owning_memory && !maybe_owning_memory->HasOwnership()) {`

I just amended the commit to have a not bugged history. But I wanted to highly the change for review.